packages/backend-common/src/sqs.ts

```diff
@@ -179,18 +191,47 @@ export const deleteMessage = async (
 	client: SQSClient,
 	queueUrl: string,
 	receiptHandle: string,
-) => {
+): Promise<DeleteResult> => {
 	try {
 		await client.send(
 			new DeleteMessageCommand({
 				QueueUrl: queueUrl,
 				ReceiptHandle: receiptHandle,
 			}),
 		);
+		return {
+			status: SQSStatus.Success,
+		};
 	} catch (error) {
-		console.error(`Failed to delete message ${receiptHandle}`, error);
-		throw error;
+		const errorMsg = `Failed to delete message ${receiptHandle}`;
+		console.error(errorMsg, error);
+		return {
+			status: SQSStatus.Failure,
+			error,
+			errorMsg,
+		};
+	}
+};
+
+export const moveMessageToDeadLetterQueue = async (
+	client: SQSClient,
+	taskQueueUrl: string,
+	deadLetterQueueUrl: string,
+	messageBody: string,
+	receiptHandle: string,
+) => {
+	// SQS doesn't seem to offer an atomic way to move message from one queue to
+	// another. There is a chance that the write to the dead letter queue
+	// succeeds but the delete from the task queue fails
+	const sendResult = await sendMessage(client, deadLetterQueueUrl, messageBody);
+	if (sendResult.status == SQSStatus.Failure) {
+		// rethrow exception, let another worker retry
+		throw Error('Failed to send message to dead letter queue');
 	}
+	// if the delete command throws an exception, it will be caught by
+	// deleteMessage and logged. Another worker will reprocess the message in
+	// the main task queue and we'll have a duplicate in the dead letter queue.
+	await deleteMessage(client, taskQueueUrl, receiptHandle);
```

Comment:
great documentation here, good shout to see if there was an atomic operation available